.github/workflows/deploy.yml

```diff
@@ -33,3 +63,41 @@ jobs:
     - name: Deploy agents
       run: |
         ./scripts/deploy.sh
+
+  test:
+    # This job only runs when a PR is merged.
+    if: github.event.pull_request.merged == true
```

Comment:
Why only merged? That's a bit too late to catch errors isn't it?

Reply:
The tests require the agents to be deployed. Since we don't want to deploy them on every PR, we have to do it this way. At least it will tell us after deployment if they are broken!

Reply:
Makes sense.